omnigibson/prims/geom_prim.py

```diff
@@ -160,6 +167,64 @@ def geom_type(self):
         """
         return self._prim.GetPrimTypeInfo().GetTypeName()
 
+    @cached_property
+    def mesh_face_centroids(self):
+        trimesh_mesh = mesh_prim_mesh_to_trimesh_mesh(
+            self.prim, include_normals=False, include_texcoord=False
+        ).convex_hull
+        assert trimesh_mesh.is_convex, f"Trying to get mesh face centroids for a non-convex mesh {self.prim_path}"
+        return th.tensor(trimesh_mesh.vertices[trimesh_mesh.faces].mean(axis=1), dtype=th.float32)
+
+    @cached_property
+    def mesh_face_normals(self):
+        trimesh_mesh = mesh_prim_mesh_to_trimesh_mesh(
+            self.prim, include_normals=False, include_texcoord=False
+        ).convex_hull
+        assert trimesh_mesh.is_convex, f"Trying to get mesh face normals for a non-convex mesh {self.prim_path}"
+        return th.tensor(trimesh_mesh.face_normals, dtype=th.float32)
+
+    def check_local_points_in_volume(self, particle_positions_in_mesh_frame):
+        mesh_type = self.prim.GetTypeName()
+        if mesh_type == "Mesh":
```

Comment:
given that the mesh type is static, maybe we can define the proper one during init / load time so we don't have to do a bunch of if / else at runtime?